src/flint/test/test_all.py

```diff
@@ -2635,9 +2662,10 @@ def setbad(obj, i, val):
         assert P([1, 1]) ** 2 == P([1, 2, 1])
         assert raises(lambda: P([1, 1]) ** -1, ValueError)
         assert raises(lambda: P([1, 1]) ** None, TypeError)
-
-        # # XXX: Not sure what this should do in general:
-        assert raises(lambda: pow(P([1, 1]), 2, 3), NotImplementedError)
+        
+        # XXX: Not sure what this should do in general:
+        # TODO: this now fails as fmpz_mod_poly allows modulus
+        # assert raises(lambda: pow(P([1, 1]), 2, 3), NotImplementedError)
```

Comment:
If we want to use 3-arg pow with poly types then we should just change this test. I'm pretty sure that I added this test and comment just as part of increasing test coverage.

I don't know if this operation makes sense for all types. It looks like `fmpz_poly` and `fmpq_poly` only have `pow_trunc` rather than `powmod`.

Maybe it would make sense to allow only special cases like:
```python
x = fmpz_poly([0, 1])
r = pow(x + 1, 100, x**3)
r = pow(x + 1, 100, 3)
```
Perhaps a `pow_trunc` method would be a good method for all types.

Reply:
I think Flint has these methods for many types, so it's definitely worth looking into. For example `fmpz_mod_poly` has `fmpz_mod_poly_pow_trunc` and `fmpz_poly` has `fmpz_poly_pow_trunc`

Reply:
is this something you want to be fixed before merging this PR?

Reply:
We should fix the test by checking for `nmod_poly` or `fmpz_mod_poly` and having a test that 3-arg `pow` works for them and raises `NotImplementedError` for other types.

We don't need to add methods to other types for this here and now. I just always want to consider what might make sense for the other types when designing any new interface/behaviour.